### PR TITLE
[FIX] event: Google & Yahoo calendar link broken

### PR DESCRIPTION
**Pull request by nboulif**
Issue

	- Install 'Event' module
	- Go to 'Event' and select any event
	- Click on "Attendees" stat button
	- Click on any attendee ( or create one if not any)
	- Confirm ( if not already confirm) registration
	- Click on "Send by email"
	- Change template to 'Event: Registration' then send
	- In the chatter, check Google or Yahoo calendar link

	Bad link.

Cause

	Wrong 'dt_format' pattern generate a broken datetime value;
	current format work with time.strftime() but not 'babel' library.

Solution

	Update 'dt_format' pattern to match with babel library since it's the
	one used in 'format_datetime' func.

 > Note that Babel uses different patterns for specifying number and date formats compared to the Python equivalents (such as time.strftime()), which have mostly been inherited from C and POSIX

ref : http://babel.pocoo.org/en/latest/dates.html#pattern-syntax

opw-2370895

### CODE REVIEW

**Comment by nboulif** [comment]
ping @nle-odoo

**Review by nle-odoo — APPROVED** [review]
in general we try not to change mail.template().body_html because then all translation become broken when someone update

in this case it might be acceptable, since there is `noupdate="1"` around it so it would only break translation for first installation of `event` module.

you should still update the `event.po` file or it will never be translated.

as a side note, before ef7871826edce400884fe032366781f59aafd8a6 (june 2019) the issue did not happen since we used `format_tz` which itself used `strftime`.

robodoo delegate+

**Comment by nboulif** [comment]
@nle-odoo Thanks for the review !
Just to be sure, can you confirm the `event.pot` changes please?

File: addons/event/i18n/event.pot [hunk 1]
```diff
@@ -178,53 +178,57 @@ msgstr ""
 #. module: event
 #: model:mail.template,body_html:event.event_subscription
 msgid ""
-"<table border=\"0\" cellpadding=\"0\" cellspacing=\"0\" style=\"padding-top: 16px; background-color: #F1F1F1; font-family:Verdana, Arial,sans-serif; color: #454748; width: 100%; border-collapse:separate;\"><tr><td align=\"center\">\n"
-"% set date_begin = format_datetime(object.event_id.date_begin, tz='UTC', dt_format='%Y%m%dT%H%M%SZ')\n"
-"% set date_end = format_datetime(object.event_id.date_end, tz='UTC', dt_format='%Y%m%dT%H%M%SZ')\n"
-"<table border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"590\" style=\"padding: 16px; background-color: white; color: #454748; border-collapse:separate;\">\n"
+"\n"
+"% if object.company_id\n"
```
Comment (nle-odoo) [inline addons/event/i18n/event.pot:182]: this part is not in the `<record id="event_subscription" model="mail.template">` content

Comment (nboulif) [reply addons/event/i18n/event.pot]: :+1:

**Comment by nboulif** [comment]
@nle-odoo Thanks again for the explanation!

robodoo r+

**Comment by landlogicit** [comment]
Hi for link Google calendar change this
% set date_begin = format_datetime(object.event_id.date_begin, tz='UTC', dt_format='yyyyMMdd\'T\'HHmmssZ')
with this:
% set date_begin = format_datetime(object.event_id.date_begin, tz='UTC', dt_format='yyyyMMdd\'T\'HHmmss')+'Z'
Similar for date_end

**Comment by nboulif** [comment]
@landlogicit Hello, may I ask you if it's to solve an issue or it's for better reading? Thank you

**Comment by nle-odoo** [comment]
Hi @landlogicit 

Thanks, it should be solved once this is merged:

https://github.com/odoo/odoo/pull/62943

**Pull request closed without merging**